homeassistant/components/binary_sensor/aurora.py

```diff
@@ -0,0 +1,152 @@
+"""
+Support for aurora forecast data sensor.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/binary_sensor.aurora/
+"""
+import logging
+from datetime import timedelta
+import requests
+import voluptuous as vol
+from homeassistant.util import Throttle
+from homeassistant.components.binary_sensor \
+    import (BinarySensorDevice, PLATFORM_SCHEMA)
+from homeassistant.const import (CONF_NAME)
+import homeassistant.helpers.config_validation as cv
+
+CONF_THRESHOLD = "forecast_threshold"
+
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_NAME = 'Aurora Visibility'
+DEFAULT_SENSOR_CLASS = "visible"
+DEFAULT_THRESHOLD = 75
+
+MIN_TIME_BETWEEN_UPDATES = timedelta(minutes=5)
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+    vol.Optional(CONF_THRESHOLD, default=DEFAULT_THRESHOLD): cv.positive_int,
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the aurora sensor."""
+    if None in (hass.config.latitude, hass.config.longitude):
+        _LOGGER.error("Lat. or long. not set in Home Assistant config")
+        return False
+
+    name = config.get(CONF_NAME)
+    threshold = config.get(CONF_THRESHOLD)
+
+    try:
+        aurora_data = AuroraData(
+            hass.config.latitude,
+            hass.config.longitude,
+            threshold
+        )
+        aurora_data.update()
+    except requests.exceptions.HTTPError as error:
+        _LOGGER.error(error)
+        return False
+
+    add_devices([AuroraSensor(aurora_data, name)], True)
+
+
+class AuroraSensor(BinarySensorDevice):
+    """Implementation of an aurora sensor."""
+
+    def __init__(self, aurora_data, name):
+        """Initialize the sensor."""
+        self.aurora_data = aurora_data
+        self._name = name
+        self.update()
```

Comment:
no, you use `add_devices(..., True)` and that make a update bevor he add to statemachine